app/bundles/CampaignBundle/Controller/ImportController.php

```
@@ -0,0 +1,280 @@
+                $this->logger->log(LogLevel::WARNING, "Import for file {$fullPath} was aborted because {$reason}.");
+
+                return $this->newAction($request, true);
+            }
+
+            if (self::STEP_UPLOAD_ZIP === $step) {
+                $valid = $this->isFormValid($form);
+
+                if ($valid) {
+                    if (file_exists($fullPath)) {
+                        unlink($fullPath);
+                    }
+
+                    $fileData = $request->files->get('campaign_import')['campaignFile'] ?? null;
+
+                    if (!empty($fileData)) {
+                        $errorMessage    = null;
+                        $errorParameters = [];
+                        try {
+                            $fs->mkdir($importDir);
+
+                            $fileData->move($importDir, $fileName);
+
+                            // $this->requestStack->getSession()->set('mautic.campaign.import.file', $fileName);
+                            $this->requestStack->getSession()->set('mautic.campaign.import.step', self::STEP_PROGRESS_BAR);
+                            // $this->requestStack->getSession()->set('mautic.campaign.import.original.file', $fileData->getClientOriginalName());
+
+                            return $this->newAction($request, true);
+                        } catch (FileException $e) {
+                            if (str_contains($e->getMessage(), 'upload_max_filesize')) {
+                                $errorMessage    = 'mautic.lead.import.filetoolarge';
+                                $errorParameters = [
+                                    '%upload_max_filesize%' => ini_get('upload_max_filesize'),
+                                ];
+                            } else {
+                                $errorMessage = 'mautic.lead.import.filenotreadable';
+                            }
+                        } catch (\Exception) {
```

Let's not catch the `\Exception` as it will catch even exceptions we want to pass on. Like the other day I was dealing with a PHPUNIT test that should have failed but it was always passing because this was in the code and it was catching the exceptions thrown by the PHPUNIT's failing assertion. We always have to catch specific exceptions. Not generic. If something fails badly, it should throw a 500 error and be logged. This will prevent knowing what's wrong during investigations.

---

fixed